torchmultimodal/models/mdetr/model.py

```diff
@@ -257,3 +260,236 @@ def mdetr_resnet101(
         extra_query_embeddings,
     )
     return mdetr
+
+
+class MultiHead(nn.Module):
+    def __init__(self, heads: nn.ModuleDict):
+        super().__init__()
+        self.heads = heads
+
+    def forward(
+        self,
+        embeddings: Tensor,
+    ) -> Dict[str, Tensor]:
+        if embeddings.size(0) != len(self.heads):
+            raise ValueError("Number of embeddings must equal number of heads")
+        out = OrderedDict()
+        for (head_name, head), embedding in zip(self.heads.items(), embeddings):
+            out[head_name] = head(embedding)
+
+        return out
+
+
+def mdetr_gqa_heads(hidden_dim: int = 256) -> MultiHead:
+    answer_type_head = nn.Linear(hidden_dim, 5)  # Number of answer types
+    answer_obj_head = nn.Linear(hidden_dim, 3)
+    answer_attr_head = nn.Linear(hidden_dim, 403)
+    answer_rel_head = nn.Linear(hidden_dim, 1594)
+    answer_global_head = nn.Linear(hidden_dim, 111)
+    answer_cat_head = nn.Linear(hidden_dim, 678)
+    heads = nn.ModuleDict(
+        {
+            "answer_type": answer_type_head,
+            "answer_obj": answer_obj_head,
+            "answer_rel": answer_rel_head,
+            "answer_attr": answer_attr_head,
+            "answer_cat": answer_cat_head,
+            "answer_global": answer_global_head,
+        }
+    )
+    return MultiHead(heads)
+
+
+class ContrastiveEmbeddingsOutput(NamedTuple):
+    query_embeddings: Tensor
+    token_embeddings: Tensor
+
+
+class MDETRVQAOutput(NamedTuple):
+    model_output: MDETRModelOutput
+    vqa_preds: Dict[str, Tensor]
+    contrastive_embeddings: ContrastiveEmbeddingsOutput
+
+
+class MDETRForVQA(nn.Module):
```

Comment:
Are those task-specific modules more suitable to be placed inside `examples/`?

Reply:
See my response to Rafi's comment 🙂 . Basically I am following the FLAVA convention here. If we do decide to keep them in core, I will add unit tests for each of these

Reply:
makes sense to put in core since people finetuning (within their code) on these will have to import from examples?

Reply:
@ebsmothers Are you going to add unit tests for those finetuning modules in a separate PR?

Reply:
I created a bootcamp task for this